### PR TITLE
Bump go version to 1.14 in go.mod

### PR DESCRIPTION
**Pull request by kakkoyun**
Signed-off-by: Kemal Akkoyun <kakkoyun@gmail.com>

I believe this has forgotten as part of https://github.com/prometheus/prometheus/pull/7100.

### CODE REVIEW

File: vendor/modules.txt [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
+## explicit
 github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-10-01/compute
 github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-10-01/network
 github.com/Azure/azure-sdk-for-go/version
 # github.com/Azure/go-autorest/autorest v0.10.0
+## explicit
 github.com/Azure/go-autorest/autorest
 github.com/Azure/go-autorest/autorest/azure
 # github.com/Azure/go-autorest/autorest/adal v0.8.3
+## explicit
```
Comment (bwplotka) [inline vendor/modules.txt:14]: uu what's this  =D

Comment (kakkoyun) [reply vendor/modules.txt:14]: This is added by `go mod vendor` using the newest go version.

**Comment by kakkoyun** [comment]
I don't know why proto generation fails. What am I missing?

**Comment by bwplotka** [comment]
Master broken? Not sure if Go 1.14 could brake anything. I know we fetch things from vendor... to build protos. You could try to run it locally @kakkoyun

**Comment by kakkoyun** [comment]
@bwplotka Yes, I'm already on it. Trying to figure it out.

**Comment by kakkoyun** [comment]
When I change the go version in `go.mod`. It downloads this `github.com/gogo/protobuf`, but it decides not to use it. It tries to use `github.com/google/protobuf`
As a result, the generator script is broken https://github.com/kakkoyun/prometheus/blob/7fb974ec92206c623ffdbe7ef445369a7782c1fd/scripts/genproto.sh#L28

And also I have realized this version acts as the minimum supported version, so if the project doesn't rely on any 1.14 features this change is unnecessary. 
I'm gonna check out if we rely on any recent features, if so I'll try to fix the issue, otherwise I'll close this one.

**Comment by kakkoyun** [comment]
Closing this for now.
> And also I have realized this version acts as the minimum supported version, so if the project doesn't rely on any 1.14 features this change is unnecessary.

**Pull request closed without merging**